apps/user-profile/src/components/modules/Onboarding/UserLandingPage/UserLandingPage.js

```diff
@@ -133,38 +121,35 @@ export default function UserLandingPage() {
     setPerson({ ...person, lastName: e.target.value, lastNameTouched: true });
   }
 
-  function onLocationChange(e) {
-    setPerson({ ...person, location: e.target.value });
+  function onDisplayNameChange(e) {
+    setPerson({
+      ...person,
+      displayName: e.target.value,
+      displayNameTouched: true,
+    });
   }
 
-  function onRoleChange(e) {
-    setPerson({ ...person, role: e.target.value });
-  }
-
-  function onHeadLineChange(e) {
-    setPerson({ ...person, headline: e.target.value });
+  function onCountryChange(e) {
+    setPerson({ ...person, country: e });
   }
 
-  function onTermsAndConditionChange(e) {
-    setPerson({ ...person, termsAndConditions: !person.termsAndConditions });
+  function onRoleChange(e) {
+    setPerson({ ...person, role: e.target.value });
   }
 
-  function onJoinNewsLetterChange() {}
-
   const onContinueClick = (e) => {
-    const profileId = userData?.profile?.id;
-    const userId = userData?.profile?.user?.id;
+    const userId = userData?.id;
 
-    if (profileId && userId) {
+    if (userId) {
       const profileRequestBody = {
-        data: {
-          displayName: `${person.firstName} ${person.lastName}`,
-          profilePictureUrl: uploadedProfilePicture || profilePicture,
-        },
+        user: userId,
+        displayName: person.displayName,
+        profilePictureUrl: uploadedProfilePicture || profilePicture,
       };
+
       setSaveInProgress(true);
 
-      agent.Profiles.put(profileId, profileRequestBody)
+      agent.Profiles.post(profileRequestBody)
```

Comment:
Is this going to create a new profile?

Reply:
Yes, we should use `post` to create a new object.

![image](https://github.com/user-attachments/assets/eca31878-0a3d-431e-b112-3c482cb56bb5)

Reply:
Sorry I mean to say, can this lead to us creating multiple profiles for the same user? Does the code handle create profile if it doesn't exist, but update if there's already one.